apis/telemetry/v1alpha1/metricpipeline_types.go

```diff
@@ -64,58 +64,57 @@ type MetricPipelineSpec struct {
 	// Configures the collection of push-based metrics that use the OpenTelemetry protocol.
-	Otlp MetricPipelineOtlpInput `json:"otlp,omitempty"`
+	Otlp *MetricPipelineOtlpInput `json:"otlp,omitempty"`
 }
 
 // MetricPipelinePrometheusInput defines the Prometheus scraping section.
 type MetricPipelinePrometheusInput struct {
 	// If enabled, Pods marked with `prometheus.io/scrape=true` annotation are scraped. The default is `false`.
-	Enabled *bool `json:"enabled,omitempty"`
+	Enabled bool `json:"enabled,omitempty"`
 	// Describes whether Prometheus metrics from specific Namespaces are selected. System Namespaces are disabled by default.
-	Namespaces MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
+	//+kubebuilder:default={exclude: {kyma-system, kube-system, istio-system, compass-system}}
+	Namespaces *MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
 }
 
 // MetricPipelineRuntimeInput defines the runtime scraping section.
 type MetricPipelineRuntimeInput struct {
 	// If enabled, workload-related Kubernetes metrics are scraped. The default is `false`.
-	Enabled *bool `json:"enabled,omitempty"`
+	Enabled bool `json:"enabled,omitempty"`
 	// Describes whether workload-related Kubernetes metrics from specific Namespaces are selected. System Namespaces are disabled by default.
-	Namespaces MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
+	//+kubebuilder:default={exclude: {kyma-system, kube-system, istio-system, compass-system}}
+	Namespaces *MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
 }
 
 // MetricPipelineIstioInput defines the Istio scraping section.
 type MetricPipelineIstioInput struct {
 	// If enabled, metrics for istio-proxy containers are scraped from Pods that have had the istio-proxy sidecar injected. The default is `false`.
-	Enabled *bool `json:"enabled,omitempty"`
+	Enabled bool `json:"enabled,omitempty"`
 	// Describes whether istio-proxy metrics from specific Namespaces are selected. System Namespaces are enabled by default.
-	Namespaces MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
+	Namespaces *MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
 }
 
 // MetricPipelineOtlpInput defines the collection of push-based metrics that use the OpenTelemetry protocol.
 type MetricPipelineOtlpInput struct {
-	// If enabled, push-based OTLP metrics are collected. The default is `true`.
-	Enabled *bool `json:"enabled,omitempty"`
-	// Describes whether push-based OTLP metrics from specific Namespaces are selected. System Namespaces are disabled by default.
-	Namespaces MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
+	// If disabled, push-based OTLP metrics are not collected. The default is `false`.
+	Disabled bool `json:"disabled,omitempty"`
+	// Describes whether push-based OTLP metrics from specific Namespaces are selected. System Namespaces are enabled by default.
+	//+optional
```

Comment:
Why is the `//+optional` marker added here but not added for the other 3 inputs?

Reply:
Agree, will add it to the rest of inputs and namespaces